src/Perpetuum/Services/MarketEngine/MarketHandler.cs

```diff
@@ -81,7 +84,17 @@ public IEnumerable<long> GetAllDefaultMarketsEids(bool includeGamma = false, boo
                 .Where(pc => includeGamma == pc.IsGammaMarket)
                 .Select(pc => pc.Market.Eid);
         }
-    
+
+        public IEnumerable<long> GetAllVisibleMarketsFor(Character c)
+        {
+            return _visibleMarkets.Get(c.Eid.ToString(), () =>
+            {
+                return _marketPriceCollectors.Values
+                .Where(pc => !pc.IsTrainingMarket)
+                .Where(pc => pc.IsVisible(c))
+                .Select(pc => pc.Market.Eid);
+            }, TimeSpan.FromMinutes(2));
```

Comment:
@TheAntiDean this is why the change did not appear immediately ;)
A natural tradeoff with caching, eventual consistency.

Or at least I don't *think* the character should have to logoff for them to see a change, only that the cached set of markets needs to expire.